src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,29 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * class that extends task
+ * labelled with an extra [D] when called
+ * stores the date which task should be done by
+ */
+public class Deadline extends Task {
+    protected LocalDate date;
+
+    public Deadline(String description, String by) {
+        super(description);
+        if (by.contains("/")) {
+            date = LocalDate.parse(by.split("/")[2] + "-" + by.split("/")[1] + "-"
+                    + (Integer.parseInt(by.split("/")[0]) < 10 ? "0" + by.split("/")[0] :
+                    by.split("/")[0]));
```

Comment:
I think the readability here can be improved. Instead of repeatedly calling the split() method on by, maybe you can consider doing
```String[] byParts = by.split("/");```
and then use byParts[2], byParts[1]... etc accordingly? I noticed the issue in several other places as well.

Reply:
Agree with^ - If appropriate, maybe you could consider extracting the parsing of the date out of your constructor or maybe do the parsing in your Parser.java? (minor OOP fix)